src/main/java/lab2/AvatarGenerator.java

```diff
@@ -1,66 +1,76 @@
-import java.awt.*;
-import java.io.IOException;
-import java.io.InputStream;
-import java.net.URI;
-import java.net.http.HttpClient;
-import java.net.http.HttpRequest;
-import java.net.http.HttpResponse;
+import javax.imageio.ImageIO; // for reading and writing images
+import javax.swing.*; // for GUI components like JFrame and JLabel
+import java.awt.*; // for user interfaces and graphics
+import java.io.IOException; // Handles input/output exceptions
+import java.io.InputStream; // For reading byte data
+import java.net.URI; // Represents a Uniform Resource Identifier
+import java.net.http.HttpClient; // For sending HTTP requests
+import java.net.http.HttpRequest; // handles an HTTP request
+import java.net.http.HttpResponse; // handles an HTTP response
 
-public class AvatarGenerator {
+public class AvatarGenerator { // AvatarGenerator class in the 'lab2' package. This class generates and displays random avatars
 
-    public static void main(String[] args) {
+    public static void main(String[] args) { // this is the entry point for running the program
 
         try {
-            var avatarStream = AvatarGenerator.getRandomAvatarStream();
-            AvatarGenerator.showAvatar(avatarStream);
-        } catch (IOException | InterruptedException e) {
-            e.printStackTrace();
-        }
 
+            var avatarStream = AvatarGenerator.getRandomAvatarStream(); // Call to getRandomAvatarStream to fetch a random avatar
+            // avatarString returns an InputStream, object reference type
+            AvatarGenerator.showAvatar(avatarStream); // Call to showAvatar with the fetched InputStream to display the avatar
+        } catch (IOException | InterruptedException e) { // Catch IOException or InterruptedException
+            e.printStackTrace(); // Print the stack trace for debugging
+        }
     }
 
     public static InputStream getRandomAvatarStream() throws IOException, InterruptedException {
-        // Pick a random style
-        String[] styles = { "adventurer", "adventurer-neutral", "avataaars", "big-ears", "big-ears-neutral", "big-smile", "bottts", "croodles", "croodles-neutral", "fun-emoji", "icons", "identicon", "initials", "lorelei", "micah", "miniavs", "open-peeps", "personas", "pixel-art", "pixel-art-neutral" };
-        var style = styles[(int)(Math.random() * styles.length)];
+        // generates a random avatar stream. Throws IOException and InterruptedException
+
+        String[] styles = { "adventurer", "adventurer-neutral", "avataaars", "big-ears", "big-ears-neutral", "big-smile"
+                , "bottts", "croodles", "croodles-neutral", "fun-emoji", "icons", "identicon", "initials", "lorelei",
+                "micah", "miniavs", "open-peeps", "personas", "pixel-art", "pixel-art-neutral" }; // styles is an array of strings
+        // Randomly selects one one avatar style from the array "styles"
+        var style = styles[(int)(Math.random() * styles.length)]; // "style" is a String
```

Comment:
And both String and String[] are reference types.
You should also menion that `Math.random()` is a class method that returns a random double between 0 and 1, and `.length` is an instance variable.